### PR TITLE
DTSRD-1205

### PR DESCRIPTION
**Pull request by andy-m-b**
### Jira link (if applicable)

https://tools.hmcts.net/jira/browse/DTSRD-1205

### Change description ###

- Add PIM for ref data sc group, to enable team members to generate user delegation SAS tokens. 

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)
- [x] Does this PR introduce a breaking change

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -164,3 +164,14 @@ variable "additional_managed_identities_access" {
...
+variable "role_assignments" {
+  type        = list(string)
+  description = "(Optional) List of roles to assign to the provided Managed Identity, scoped to this storage account."
+  default     = [
+    "Storage Account Delegator",
+    "Storage Blob Delegator",
+    "Storage Blob Data Contributor",
+    "Storage Blob Data Reader"
+  ]
+}
```
Comment (timja) [inline variables.tf:177]: this is unused

File: storage-account-rd-data-extracts.tf [hunk 2]
```diff
@@ -2,6 +2,21 @@ locals {
   rd_data_extract_product         = "rddataextract"
   rd_data_extract_account_name    = join("", [local.rd_data_extract_product, var.env])
   rd_data_extract_container_name  = "rd-data-extract"
+
+  allowed_roles = [
+    "Storage Account Delegator",
```
Comment (timja) [inline storage-account-rd-data-extracts.tf:7]: Storage Account Delegator isn't a role: https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles
```suggestion
    "Storage Account Delegator",
```

File: storage-account-rd-data-extracts.tf [hunk 3]
```diff
@@ -17,12 +32,20 @@ module "storage_account_rd_data_extract" {
 
   enable_https_traffic_only = true
 
+  pim_roles = {
+    "Storage Account Delegator" = {
+      principal_id = data.azuread_group.sc_group.id
+    }
```
Comment (timja) [inline storage-account-rd-data-extracts.tf:38]:
```suggestion
```

**Review by timja — COMMENTED** [review]
Do you need these as well?

* Storage Blob Data Contributor
* Storage Blob Data Reader

File: storage-account-rd-data-extracts.tf [hunk 4]
```diff
@@ -2,6 +2,22 @@ locals {
...
+
+  allowed_roles = [
+    "Storage Blob Data Contributor",
+    "Storage Blob Delegator",
+    "Storage Blob Data Reader"
+  ]
+}
+
+data "azuread_group" "sc_group" {
+  display_name     = "DTS Ref Data SC (env:production)"
```
Comment (timja) [inline storage-account-rd-data-extracts.tf:14]: we don't generally put an env in a team group like this but it should be fine, I can see it's created, (although there's no one in there)

File: storage-account-rd-data-extracts.tf [hunk 5]
```diff
@@ -2,6 +2,22 @@ locals {
   rd_data_extract_product         = "rddataextract"
   rd_data_extract_account_name    = join("", [local.rd_data_extract_product, var.env])
   rd_data_extract_container_name  = "rd-data-extract"
+
+  allowed_roles = [
+    "Storage Blob Data Contributor",
+    "Storage Blob Delegator",
+    "Storage Blob Data Reader"
+  ]
```
Comment (timja) [inline storage-account-rd-data-extracts.tf:10]: unused
```suggestion
```

File: storage-account.tf [hunk 6]
```diff
@@ -38,6 +38,9 @@ module "storage_account" {
 
   sa_subnets = local.all_valid_subnets
 
+  network_rules {
+    ip_rules                   = var.ip_rules
+  }
```
Comment (timja) [inline storage-account.tf:43]:
```suggestion
    ip_rules                   = var.ip_rules
```

**Review by sahitya-desireddy — APPROVED** [review]

**Review by manukundloo-hmcts — APPROVED** [review]

File: storage-account-rd-data-extracts.tf [hunk 7]
```diff
@@ -2,6 +2,17 @@ locals {
   rd_data_extract_product         = "rddataextract"
   rd_data_extract_account_name    = join("", [local.rd_data_extract_product, var.env])
   rd_data_extract_container_name  = "rd-data-extract"
+
```
Comment (timja) [inline storage-account-rd-data-extracts.tf:5]:
```suggestion
```

File: storage-account-rd-data-extracts.tf [hunk 8]
```diff
@@ -4,6 +4,16 @@ locals {
...
 
+data "azuread_group" "sc_group" {
+  display_name     = "DTS Ref Data SC (env:production)"
+  security_enabled = true
+}
+
+data "azurerm_client_config" "this" {}
+
+data "azurerm_subscription" "primary" {
+}
```
Comment (timja) [inline storage-account-rd-data-extracts.tf:15]:
```suggestion
```

File: storage-account-rd-data-extracts.tf [hunk 9]
```diff
@@ -4,6 +4,12 @@ locals {
   rd_data_extract_container_name  = "rd-data-extract"
 }
 
+data "azuread_group" "sc_group" {
+  display_name     = "DTS Ref Data SC (env:production)"
+  security_enabled = true
+}
+
+
```
Comment (timja) [inline storage-account-rd-data-extracts.tf:12]:
```suggestion
```

**Review by timja — APPROVED** [review]

**Review by sahitya-desireddy — APPROVED** [review]

File: storage-account-rd-data-extracts.tf (hunk 4 above)
Comment (andy-m-b) [reply storage-account-rd-data-extracts.tf:8]: I will remove this env variable and create another PR.

**Pull request merged**